scalyr_agent/log_processing.py

```diff
@@ -2282,6 +2322,46 @@ def config(self):
         """
         return self.__log_entry_config
 
+    def finish( self ):
```

Comment:
If we are moving the names of this to `close` in `LogProcessor` we should probably call it `close` in the `LogMatcher` as well.  So, maybe `close_when_processing_completed` (ok, don't hit me).  I hate to just use `close` here since the name to me implies the `LogProcessor` will be closed when the method finishes (it is synchronous).  But, I can't up with a better name then `close_when_processing_completed`